### PR TITLE
python3.pkgs.umap-learn: 0.4.5 -> 0.5.0 (fix the build)

### PR DESCRIPTION
**Pull request by Mic92**
###### Motivation for this change

required for the next TTS release

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pynndescent/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  pname = "pynndescent";
+  version = "0.5.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "74a05a54d13573a38878781d44812ac6df97d8762a56f9bb5dd87a99911820fe";
+  };
+
+  patches = [
+    # fixes tests, included in next version
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pynndescent/default.nix:23]:
```suggestion
    # fixes tests, included in 0.5.2
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 110688` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.scikit-tda</li>
    <li>python39Packages.umap-learn</li>
  </ul>
</details>
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>python37Packages.scikit-tda</li>
    <li>python37Packages.umap-learn</li>
    <li>python38Packages.scikit-tda</li>
    <li>python38Packages.umap-learn</li>
    <li>python39Packages.pynndescent</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python37Packages.pynndescent</li>
    <li>python38Packages.pynndescent</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110688` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>python39Packages.scikit-tda</li>
<li>python39Packages.umap-learn</li>
</ul>
</details>
<details>

  <summary>7 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.pynndescent: log was empty</li>
<li>python37Packages.scikit-tda: log was empty</li>
<li>python37Packages.umap-learn: log was empty</li>
<li>python38Packages.pynndescent: log was empty</li>
<li>python38Packages.scikit-tda: log was empty</li>
<li>python38Packages.umap-learn: log was empty</li>
<li>python39Packages.pynndescent: log was empty</li>
  </ul>
</details>

Segfaults in sandbox

**Comment by Mic92** [comment]
@GrahamcOfBorg build python38Packages.umap-learn python37Packages.umap-learn

**Comment by Mic92** [comment]
@GrahamcOfBorg build python38Packages.umap-learn python37Packages.umap-learn

**Comment by Mic92** [comment]
> Result of `nixpkgs-review pr 110688` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 2 packages marked as broken and skipped:
> 5 packages failed to build:
> 
>     * python37Packages.scikit-tda
> 
>     * python37Packages.umap-learn
> 
>     * python38Packages.scikit-tda
> 
>     * python38Packages.umap-learn
> 
>     * python39Packages.pynndescent
> 
> 
> 2 packages built:
> 
>     * python37Packages.pynndescent
> 
>     * python38Packages.pynndescent

Did you skipped building tensorflow here, I built it on my machine.

**Comment by SuperSandro2000** [comment]
> Did you skipped building tensorflow here, I built it on my machine.

Yes I do.

> does not build all packages (e.g. lumo, tensorflow or pytorch).

**Comment by Mic92** [comment]
@GrahamcOfBorg build python38Packages.umap-learn

**Pull request merged**